src/controllers/wordlist-controller.js

```diff
@@ -1,122 +1,223 @@
-import { PsEvent, LanguageModelFactory as LMF, Constants, Lemma, Inflection, DefinitionSet, Lexeme, Homonym } from 'alpheios-data-models'
+import { PsEvent } from 'alpheios-data-models'
 import WordList from '@/lib/word-list'
-import IndexedDBAdapter from '@/storage/indexed-db-adapter'
+import WordItem from '@/lib/word-item'
 
 export default class WordlistController {
-  constructor (userID) {
-    this.userID = userID
+  /**
+   * @constructor
+   * @param {String[]} availableLangs language codes
+   * @param {PSEvent[]} events events that the controller can subscribe to
+   */
+  constructor (availableLangs,events) {
     this.wordLists = {}
-
-    this.storageAdapter = new IndexedDBAdapter()
+    this.availableLangs = availableLangs
+    events.TEXT_QUOTE_SELECTOR_RECEIVED.sub(this.onTextQuoteSelectorReceived.bind(this))
+    events.HOMONYM_READY.sub(this.onHomonymReady.bind(this))
+    events.DEFS_READY.sub(this.onDefinitionsReady.bind(this))
+    events.LEMMA_TRANSL_READY.sub(this.onLemmaTranslationsReady.bind(this))
   }
 
-  get availableLangs () {
-    return ['lat', 'grc', 'ara', 'per', 'gez']
+  /**
+   * Asynchronously initialize the word lists managed by this controller
+   * @param {UserDataManager} dataManager a user data manager to retrieve initial wordlist data from
+   *  // TODO may need a way to process a queue of pending words here e.g. if the wordlist controller isn't
+    * // activated until after number of lookups have already occurred
+   * Emits a WORDLIST_UPDATED event when the wordlists are available
+   */
+  async initLists (dataManager) {
+    for (let languageCode of this.availableLangs) {
+      let wordItems = await dataManager.query({dataType: 'WordItem', params: {languageCode: languageCode}})
```

Comment:
I think ideally, parallel requests make sense here, since order doesn't matter, but in reality, users are likely to have lists for only 1 or 2 languages at most, so I am not sure it's a critical optimization at the moment.